src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -442,8 +452,12 @@ public boolean store(Tangle tangle, Snapshot initialSnapshot) throws Exception {
         // non-cached operations.
         List<Pair<Indexable, Persistable>> batch = getSaveBatch();
         cacheApprovees(tangle);
-        if (tangle.getCache(TransactionViewModel.class).get(hash) == null) {
-            cachePut(tangle, this, hash);
+
+        Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
+        if (cache != null) {
+            if (cache.get(hash) == null) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/GalRogozinski/iri/pullRequest?prid=4804352)